src/UI/Seller/src/app/sellers/components/locations/seller-location-edit/seller-location-edit.component.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  OnChanges,
+} from '@angular/core'
+import { FormGroup, Validators, FormControl } from '@angular/forms'
+import { Address, ListPage } from '@ordercloud/angular-sdk'
+import { MiddlewareAPIService } from '@app-seller/shared/services/middleware-api/middleware-api.service'
+import { ActivatedRoute } from '@angular/router'
+import { GeographyConfig } from '@app-seller/shared/models/supported-countries.constant'
+import {
+  ValidateCAZip,
+  ValidateUSZip,
+  ValidatePhone,
+} from '@app-seller/validators/validators'
+import { takeWhile } from 'rxjs/operators'
+import { SellerAddressService } from '../seller-address.service'
+import {
+  OrderCloudIntegrationsConversionRate,
+  HeadStartSDK,
+} from '@ordercloud/headstart-sdk'
+import { SupportedCountries } from '@app-seller/models/currency-geography.types'
+
+@Component({
+  selector: 'app-seller-location-edit',
+  templateUrl: './seller-location-edit.component.html',
+  styleUrls: ['./seller-location-edit.component.scss'],
+})
+export class SellerLocationEditComponent implements OnChanges {
+  alive = true
+  locationHasNoProducts: boolean
+  countryOptions: SupportedCountries[]
+  flag: string
+  resourceForm: FormGroup
+  zipLabel = 'Zip Code'
+  zipPlaceholder = 'Enter Zip'
+  stateLabel = 'State'
+  statePlaceholder = 'Enter State'
+  cityLabel = 'City'
+  cityPlaceholder = 'Enter City'
+  street1Label = 'Street 1'
+  street1Placeholder = 'Enter Street 1'
+  countryHasBeenSelected = false
+  currencyOptions: OrderCloudIntegrationsConversionRate[]
+  @Input()
+  set location(sellerLocation: Address) {
+    if (sellerLocation.ID) {
+      this.handleSelectedLocationChange(sellerLocation)
+    } else {
+      this.handleSelectedLocationChange(
+        this.sellerLocationService.emptyResource
+      )
+    }
+  }
+  @Input()
+  filterConfig
+  @Input()
+  suggestedAddresses: ListPage<Address>
+  @Output()
+  updateResource = new EventEmitter<any>()
+  @Output()
+  selectAddress = new EventEmitter<any>()
+  @Output()
+  canDelete = new EventEmitter<boolean>()
+
+  constructor(
+    private middleware: MiddlewareAPIService,
+    private activatedRoute: ActivatedRoute,
+    private sellerLocationService: SellerAddressService
+  ) {
+    this.countryOptions = GeographyConfig.getCountries()
+  }
+
+  ngOnChanges() {
+    this.setLocationID()
+  }
+
+  async handleSelectedLocationChange(location: Address): Promise<void> {
+    // Bug with the SDK where GetRateList has wrong return type
```

Comment:
If the SDK is now local can't this be fixed in this PR?

Reply:
For this one, I am not sure what the bug was or if it has been fixed.  That is a comment I carried over from the Supplier logic.

Reply:
Does it work if you remove the try/finally? Like this:
```typescript
HeadStartSDK.ExchangeRates.GetRateList().then((res) => {
    this.currencyOptions = res.Items
    location.Country && this.setFlag(location.Country)
})
this.createSellerLocationForm(location)
```
If so, can you you rewrite it and remove the comment? Its possible this was fixed.

Reply:
It appeared to work that way, so I updated it.